### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by Jdochoa** [mariuz/flamerobin#216] [Ideas]
Hi all,
This is the list of the features added to FB and the status they currently have.
Please review and make your comments about them and / or new ones to add. 

    

- **Database:**
     - [X]  ~~SQL Security.~~ 
     - [X]  ~~Linger.~~ 
     - [ ]  Replication.

- **Procedures:**
     - [X]  ~~_Hide procedures from packages._~~
     - [X]  ~~_UDR functionality._~~ 
     - [X]  ~~_SQL Security._~~

- **Functions:**
     - [X]  ~~_Hide functions from packages._~~
     - [X]  ~~_UDR functionality._~~
     - [X]  ~~_Separate UDF._~~
     - [X]  ~~_PSQL Functions._~~
     - [X]  ~~_SQL Security._~~

- **Domains:**
     - [X]  ~~_separate sysdomain._~~

- **Triggers:**
     - [X]  ~~_Separate DDL Triggers._~~
     - [X]  ~~_Separate DB Triggers._~~
     - [X]  ~~_SQL Security._~~

- **Tables:**
     - [X]  ~~_Separatte GTT._~~
     - [X]  ~~_SQL Security._~~
     - [ ]  Agregate Identity columns. (In progress)
     - [ ]  Replication. 

- **Packages:**
     - [X]  ~~_Basic functionality._~~
     - [X]  ~~_SQL Security._~~
     - [X]  ~~_Agregate syspackages._~~

- **Index:**
     - [ ]  Add index node. (In Progress)

- **Users:**
     - [ ]  Add users node. 

- **Firebird 4**
     - [X] ~~_DataTypes._~~

- **Misc**
     - [ ]  Icons.  (In Progress)
     - [ ]  FB Services.
     - [X] ~~_Shutdown/Startup Database._~~
     - [ ]  Configuration file manager.

**Comment by arvanus** [comment]
I think that Replication can be ~removed~ for now, because it depends heavily in the databases.conf file, at least I hadn't heard of a possibility to not use it right know

**Reply by luronumen** [reply]
I agree too!

**Reply by Jdochoa** [reply]
> I think that Replication can be ~removed~ for now, because it depends heavily in the databases.conf file, at least I hadn't heard of a possibility to not use it right know

This is not true, please see https://github.com/FirebirdSQL/firebird/blob/master/doc/README.replication.md

./jo

**Comment by arvanus** [comment]
In the Database->Linger, needs to implement the service part, to change it too, not only read it's value
https://firebirdsql.org/file/documentation/release_notes/html/en/3_0/rnfb30-ddl-enhance.html#rnfb30-ddl-enhance-linger
See #217
